app/Controllers/Transactions.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Controllers;
+
+use App\Models\PaymentModel;
+
+
+
+
+class Transactions extends BaseController
+{
+//Function to view tenant's payment history
+	public function tenantHistory($tenantID)
+	{
+		$db = db_connect();
+		$model = new PaymentModel($db);
+
+		$tenant = $model->isTenant($tenantID);
+		
+		if ($tenant == true) {
+			
+	
+			$result = $model->getTenantHistory($tenantID);
+
+
+			$data = ['data' => $result];
+		
+
+		}
+		else{
+			$data = ['data' => 'User not a Tenant'];
+		}
+
+
+		return json_encode($data);
+
+	}
+
+//Function to make payments
+	public function makePayment($tenantID)
+	{
+
+		$db = db_connect();
+		$model = new PaymentModel($db);
+
+
+
+		$property = $model->getProperty($tenantID);
+
+/*		echo "<pre>";
+		print_r($property);
+		echo "</pre>";*/
+
+
+		
+
+		if (isset($_POST["Submit"])){
```

Comment:
This won't necessarily be a form being submitted. Look at the code below for how a maintainance request is added and use a similar approach for all `POST `requests you're handling

<details>
<summary>CODE</summary>

 ```php
 $request = new RequestsModel();

if($this->request->getMethod() == 'post'){
            $json = $this->request->getJSON();
            $data = [
                'propertyID' => $json->propertyID,
                'requestMessage' => $json->requestMessage,
            ];
    
            if($request->insert($data)){
                return true;
            }
            else{
                return false;
            }
        }
```
</details>

> **_NOTE:_**  Make sure you can insert/update data in your local database from postman before submitting any code handling a POST request